### DISCUSSION TITLE
Is it possible to doing something similar for NDS/DSi mode on 3DS?

### DISCUSSION THREAD

**Discussion by Gregoras** [profi200/open_agb_firm#193] [General]
I know, there is Twillight Menu++, but it has many issues. I.e. loading time and crap load of outdated emulators. And after all TWM++ is just pirate cartridge firmware forcing somehow to run on 3DS.

And you found the way to makes 3DS one of the best proper (so not a emulator) GBA on the market. So I wonder if similar thing with NDS/DSi firmware could be possible? There was NTRLDR, but was never heavy developed.

**Comment by profi200** [comment]
Not sure to be honest. If you mean the color correction then the value for DS games is barely there. The vast majority of games is optimized for the DS lite LCDs. If you want to do it anyway you should have to know that the CPU is doing the conversion via a lookup table and this costs memory bandwidth + power. The highly optimized assembly i wrote is probably fast enough to handle 2 frames but it's on the edge of what we can do on the hardware. And we would have to sacrifice on accuracy because the lookup table is already large for 5 bits. DS video uses 6 bits per channel.

That being said there is a way to improve DS mode which is tweaking the gamma curves to match the LCDs better which is what i'm working on. You may have noticed that DS games look rather washed out on 3DS and this should fix it.

**Reply by csteingraber** [reply]
Out of curiosity, was this work of tweaking the gamma curves to reduce the washed out tendency of DS games ever incorporated into something like twilight menu or TWPatcher?

**Comment by Gregoras** [comment]
By writing "similar thing for NDS" I mean running NDS/DSi firmware, preferably with home menu, in separate mode, directly from boot, with access to cartridge port.

**Comment by profi200** [comment]
That's not something i'm working on but at least 1 other person is working on the TWL counterpart. However it's more of a drop-in replacement than a FIRM you can just boot from cold boot. TWL_FIRM is much more complicated than AGB_FIRM.
